packages/plugins/graphql-jit/src/index.ts

```diff
@@ -89,72 +83,77 @@ export const useGraphQlJit = (
       }
 
       jitCacheByDocument.set(args.document, cacheEntry);
-      if (documentSource) {
-        jitCacheByDocumentString.set(documentSource, cacheEntry);
+      if (jitCacheByDocumentString) {
+        const documentSource = getDocumentString(args.document);
+        if (documentSource) {
+          jitCacheByDocumentString.set(documentSource, cacheEntry);
+        }
       }
     }
     return cacheEntry;
   }
 
+  function jitExecutor(args: TypedExecutionArgs<unknown>) {
+    const cacheEntry = getCacheEntry(args);
+
+    const result$ = cacheEntry.subscribe
+      ? (cacheEntry.subscribe(
+          args.rootValue,
+          args.contextValue,
+          args.variableValues,
+        ) as PromiseOrValue<ExecutionResultWithSerializer>)
+      : (cacheEntry.query(
+          args.rootValue,
+          args.contextValue,
+          args.variableValues,
+        ) as PromiseOrValue<ExecutionResultWithSerializer>);
+    if (isPromise(result$)) {
+      return result$.then(r => {
+        r.stringify = cacheEntry.stringify;
+        return r;
+      });
+    }
+    result$.stringify = cacheEntry.stringify;
+    return result$;
```

Comment:
Since this pattern happens a lot, perhaps we can make an util function with a name describing what it do ? Because the syntax noise obfuscate a bit the logic i think.

Something like 
```ts
function handlePromiseOrValue<P, R>(p: PromiseOrValue<P>, handler): PromiseOrValue<R> {
	if(isPromise(p) {
		return p.then(value => handler(value))
	}
	return handler(p)
}

// which would lead to this usage here:
	return handlePromiseOrValue(result$, result => {
		result.stringify = cacheEntry.stringify
		return result
	})
```

Reply:
Actually we can use `value-or-promise` package.